WordPress/Classes/Extensions/UIViewController+Helpers.swift

```diff
@@ -10,4 +12,16 @@ extension UIViewController
         
         return visibleAsRoot || visibleAsTopOnStack || visibleAsPresented
     }
+    
+    // Determines if the current ViewController is being presented modally
+    //
+    public func isModal() -> Bool {
+        return self.presentingViewController != nil
+    }
+    
+    // Determines if the current ViewController is the only one in the Navigation stack
+    //
+    public func isRootInNavigation() -> Bool {
+        return navigationController?.viewControllers.count == 1
+    }
```

Comment:
I've been on the fence about this for a very long time. When I see this methods something feels wrong: a VC shouldn't be querying how it's presented to figure out how to dismiss itself. And in a way, I think the problem is that a VC shouldn't be dismissing itself at all.

> func dismissViewControllerAnimated(_ flag: Bool, completion completion: (() -> Void)?)
> The presenting view controller is responsible for dismissing the view controller it presented. If you call this method on the presented view controller itself, UIKit asks the presenting view controller to handle the dismissal.

It works when you call dismiss from the presented VC, but the API looks like it was meant to be called from the presenter. We have some instances of this in Login/Stats with a `dismissBlock`.

However, I'm not sure that's right either, since it's the presented VC the one adding a "Close" button, so it should be able to close itself right?

Dunno, I just feel that peeking at the navigation stack to make decisions is a bit like playing with fire, and you'll eventually get burned, but I don't really have a solution :disappointed: